File: SlideTwitter/src/main/java/tech/slideshare/twitter/database/TweetQueueDao.java
package tech.slideshare.twitter.database;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class TweetQueueDao extends AbstractDao {

    public TweetQueueDao(Connection con) {
        super(con);
    }

    public boolean delete(int slideId) throws SQLException {
        String sql = "DELETE FROM tweet_queue WHERE slide_id = ?";
        try (PreparedStatement pstmt = con.prepareStatement(sql)) {
            pstmt.setInt(1, slideId);

            return pstmt.executeUpdate() > 0;
        }
    }
}

File: SlideTwitter/src/main/java/tech/slideshare/twitter/database/TweetQueueDao.kt
package tech.slideshare.twitter.database

import java.sql.Connection

class TweetQueueDao(con: Connection) : AbstractDao(con) {

    fun delete(slideId: Int): Boolean {
        val sql = "DELETE FROM tweet_queue WHERE slide_id = ?"

        con.prepareStatement(sql).use {
            it.setInt(1, slideId)

            return it.executeUpdate() > 0
        }
    }
}
